Mathport/Syntax/Translate/Basic.lean

```diff
@@ -619,8 +618,11 @@ mutual
     | Tactic.«<|>» tacs => do
       `(tactic| first $[| $(← tacs.mapM fun tac => trTactic tac):tactic]*)
     | Tactic.«[]» _tacs => warn! "unsupported (impossible)"
-    | Tactic.exact_shortcut ⟨_, Expr.calc args⟩ => do
-      `(tactic| calc $(← trCalcArgs args)*)
+    | Tactic.exact_shortcut ⟨m, Expr.calc args⟩ => withSpanS m do
+      if h : args.size > 0 then
+        `(tactic| calc $(← trCalcArg args[0]) $(← args[1:].toArray.mapM trCalcArg)*)
+      else
+        warn! "empty calc is unsupported"
```

Comment:
```suggestion
        warn! "unsupported (impossible)"
```

Reply:
I tend to put in more descriptive error messages so that it's easier to debug when the impossible happens, as it often tends to do.  (Every build, mathport encounters roughly 500 impossibilities.)

But I'm happy to change this for consistency.

Reply:
The messages should come with line:col numbers for debugging in that case. Using `(impossible)` is important information to explain whether something is unsupported by mathport or unsupported by the lean 3 export format.

Obviously any `(impossible)` occurrences in the actual output should be debugged: they used to be panics and we should have zero of those in mathport.

Reply:
Could you report the (deduplicated) list of impossibles that occur on mathlib as an issue? With an exemplar definition for each occurrence as well, if possible.

Reply:
#158 #159 maybe I missed some